### PR TITLE
Fixes a bug that prevented classname from being set on web components

### PR DESCRIPTION
**Pull request by briandeheus**
During development on our platform we found out that we could not set the classname on a web component.

Here is a jsfiddle demonstrating the bug: https://jsfiddle.net/zehydcnq/

What causes this bug to occur is that ReactDOM sets properties on web components using setValueForAttribute but never setValueForProperty, which does some marshalling with the className property.

My change adds an extra check to see whether a property is part of the DOMProperties and if so to set the property as opposed to the attribute.

I've made sure to add add/update/remove tests.

### CODE REVIEW

**Comment by briandeheus** [comment]
Not sure why Travis fails to server side rendering test. On my machine™ it works just fine.

**Comment by gaearon** [comment]
Hi, thanks for the PR!

This is not a bug but rather intentional behavior. Please see https://github.com/facebook/react/issues/4933#issuecomment-145951830 for the rationale.

TLDR: Use `class` on Web Components instead.

**Comment by briandeheus** [comment]
Fair enough! Thanks!
